### ISSUE TITLE
Bruteforce Failed

### ISSUE THREAD

**Issue by ghost** [bao7uo/dp_crypto#1]
I'm trying this tool for version **2017.3.913** but wasn't successful. I have tried several combinations for the "key_length", "key_charset", and "accuracy", but it still didn't work. Any help?

**Comment by bao7uo** [comment]
Sorry for the late response @phvin . If the key has been set up securely then it won't always work. It works best when they have just installed the Telerik components and not changed the defaults.

**Comment by h00die** [comment]
Can you clarify the vulnerable versions?
[EDB](https://www.exploit-db.com/exploits/43873) shows **< 2017.1.118**
[github](https://github.com/bao7uo/dp_crypto/blob/master/dp_crypto.py#L242) shows **<=2017.3.913**
[telerik](https://www.telerik.com/support/kb/aspnet-ajax/details/cryptographic-weakness#full-patches) implies **<=2017.2.503**
[CVE](https://nvd.nist.gov/vuln/detail/CVE-2017-9248) shows **<R2 2017 SP1** which [translates](https://www.telerik.com/support/whats-new/aspnet-ajax/release-history/ui-for-asp-net-ajax-r2-2017-sp1-(version-2017-2-621)) **<2017.2.621**

The telerik and cve writeups seem to align with each other.  Any insights?

**Comment by bao7uo** [comment]
EDB shows thru 2017.1.118 which means <= 2017.1.118. Github doesn't show that 2017.3.913 is vulnerable, it is just providing the exploit with the capability to recognise and check those versions.

I know that 2017.1.118 version is vulnerable because I have tested the exploit on it. I've also tested on 2017.2.621 and 2017.2.711 and they are not vulnerable. I don't think I have tested on 2017.2.503.

I agree with you that Telerik and the CVE indicate that 2017.2.503 is vulnerable. Sometimes it can take a few attempts to get the exploit to work.

Is the server running a foreign language version of Windows? This can affect exploitability. I am planning to fix this at some point.
